index.js

```diff
@@ -1,135 +1,35 @@
-            default:
-                break;
-        }
-
-        console.log('results:', result);
+async function docable(argv, report) {
+    let stepper = new Stepper(argv.doc, argv.html);
+    await stepper.setup(argv.doc, argv.html);
+    const { $, results } = await stepper.run();
 
-        if (result.exitCode === 0) {
-            result.status = true;
-            stepper._setResults($(cell.elem).parent('pre'), result);
-        }
-        else {
-            result.status = false;
-            stepper._setResults($(cell.elem).parent('pre'), result);
-        }
-
-    }
-
-    await fs.promises.writeFile(path.join(path.dirname(notebookPath), 'notebook_results.html'), $.html(), { encoding: 'utf-8' });
-}
-
-async function testreport(mode, argv, options = {rendered: undefined, selector: undefined, css: undefined, textSelector: undefined})
-{
-    // documents and associated steps; connector to infrastructure provider
-    let stepper = new Steps(options.renderer, options.selector, options.css);
-
-    let {docs, conn, cwd, targets, clean, verify} = await stepper.read(argv.stepfile);
-
-    console.log(`Using cwd ${cwd}`);
-    let op = new Operators(conn, cwd,targets);
-    let sl = new Select( op, options.textSelector );
-
-    console.log(chalk`{bold \nRunning documentation tests:\n}`)
-
-    // Select/translate/perform/assert workflow
-    
-    let results_dir = path.join(path.dirname(argv.stepfile), 'docable_results');
-    if (!fs.existsSync(results_dir)) {
-        fs.mkdirSync(results_dir);
+    if (report) {
+        const reporter = new Reporter($, results);
+        let docStatus = await reporter.report();
+        process.exitCode = docStatus ? 0 : 1;
     }
 
-    let results = [];
-    for( let doc of docs )
-    {
-        let engine = doc.engine;
-        for( let stepFn of doc.steps )
-        {
-            let result = await stepFn(engine, sl);
-            results.push(result);
-        }
-
-        if (verify) {
-            verifyOut = await op.run(verify);
-            console.log(chalk`{${verifyOut.exitCode == 0 ? 'green' : 'red'} ${verifyOut.stdout + '\n' + verifyOut.stderr}}`);
-            doc.engine('body').append(`<h2>docable verification results</h2><div class="verify ${verifyOut.exitCode == 0 ? 'passing' : 'failing'}">$ ${verify}\n${verifyOut.stdout}\n${verifyOut.stderr}</div>`);
-        }
-
-        if (mode == "report") {
-            let reportHtml = path.join(results_dir, path.basename(doc.file, '.md') + '.html');
-            fs.writeFileSync(reportHtml, doc.engine.html())
-            console.log(`Generated report ${reportHtml}`);
-        }
-    }
-
-    // Close spawned processes
-    console.log('cleaning up...');
-    await op.tearDown(targets);
-    await op.run(clean);
-    
-
-    let exitCode = 0;
-    if(results.filter(result => result.status == false).length > 0) exitCode = 1;
+    // TODO:
+    // await stepper.tearDown();
 
-    process.exit(exitCode);
+    return results;
```

Comment:
This seems to be mixed with the concern of conditionally existing with exit code or not, based on being called from a command line option.

Perhaps the right fix is to appropriately make a module version to call that is distinct from cli.

For example, in dockerizeme:

### CLI

To use DockerizeMe to dockerize an application
```
dockerizeme run --verbose <file|application dir>
```
### Module

As a module

```
const dockerizeme = require('dockerizeme');
(async () => {
    
    let options = { pkg: '<application dir>' };
    let contents = await dockerizeme.run(options);
    console.log(contents);
    
})();
```

Reply:
Handled with updated method for exitcode/results.